src/main/scala/ir/Statement.scala

```diff
@@ -128,7 +130,40 @@ object GoTo:
 
 sealed trait Call extends Jump
 
-class DirectCall(val target: Procedure, var returnTarget: Option[Block],  override val label: Option[String] = None) extends Call {
+abstract trait FallThrough extends HasParent[Block]:
```

Comment:
why does this need to be separate from Call? `abstract trait` also is not really anything, it isn't any different to `trait`. @ailrst

Reply:
Its a clear single piece of functionality, the invariant (parent.fallthrough == this.returntarget, if and when it is set), its separate so its responsibility is clearer. You could also conceivably have a Call that cannot have a fallthrough, e.g nonterminating depending on how we represent that

Reply:
Do you mean non-returning calls? They aren't currently separate in the IR and if they were then Fallthrough shouldn't have an Option at all. I don't see the point in keeping things separate for hypothetical future refactors that we have no idea if we even will want to do, and Call does not have any functionality associated with it at present anyway.

Reply:
I wrote it that way because I thought the code was clearer, if you disagree you can change it however you like.